apisix/plugin.lua

```diff
@@ -901,7 +961,15 @@ _M.stream_check_schema = stream_check_schema
 
 function _M.plugin_checker(item, schema_type)
     if item.plugins then
-        return check_schema(item.plugins, schema_type, true)
+        local ok, err = check_schema(item.plugins, schema_type, true)
+
+        if ok then
```

Comment:
Add judge `data_encryption == true` here ? if `data_encryption == false`, no need to iterate the plugin loop

Reply:
would check `data_encryption == true` in `decrypt_conf` function, ref: https://github.com/apache/apisix/pull/8403/files/6fd1700dd37fc68d4b057d302229c06bf342eb24#diff-3936316fb5318f20b79ffe05648cf3d6b8c61e6987b51b52e2571c8a8e73a68bR867-R870

Reply:
But still need to iterate the loop below, we can skip the iteration
```
            for name, conf in pairs(item.plugins) do
                decrypt_conf(name, conf, schema_type)
            end
```

Reply:
updated